src/RichTextEditor/utilities/toolbarHelpers.js

```diff
@@ -16,25 +18,31 @@ export const isMarkActive = (editor, format) => {
 
 export const toggleBlock = (editor, format) => {
   const isActive = isBlockActive(editor, format);
-  const isList = SCHEMA.LIST_TYPES.includes(format);
-
-  Transforms.unwrapNodes(editor, {
-    match: n => SCHEMA.LIST_TYPES.includes(n.type),
-    split: true,
-  });
-
-  /* eslint no-nested-ternary: 0 */
-  Transforms.setNodes(editor, {
-    type: isActive
-      ? SCHEMA.PARAGRAPH
-      : isList
-        ? SCHEMA.LIST_ITEM
-        : format,
-  });
-
-  if (!isActive && isList) {
-    const block = { type: format, children: [], data: { tight: true } };
-    Transforms.wrapNodes(editor, block);
+  const isList = input => LIST_TYPES.includes(input);
+  const isQuote = input => input === BLOCK_QUOTE;
+  const isListItem = input => input === LIST_ITEM;
+
+  /* Clear selection of block types */
+  Transforms.unwrapNodes(editor, { match: n => isQuote(n.type), split: true });
+  Transforms.unwrapNodes(editor, { match: n => isListItem(n.type), split: true });
+  Transforms.unwrapNodes(editor, { match: n => LIST_TYPES.includes(n.type), split: true });
+
+  if (!isActive) {
+    const formattedBlock = { type: format, children: [], data: { tight: true } };
```

Comment:
Don't we only need the `tight` property if it's a list? I don't _think_ we need it if it's a block quote.

Reply:
I didn't see a change when doing so, but I made this be an empty object if it's a block quote.